addons/website/models/website.py

```diff
@@ -183,10 +184,19 @@ def _bootstrap_homepage(self):
         </t>''' % (self.id)
         standard_homepage.with_context(website_id=self.id).arch_db = new_homepage_view
 
-        self.homepage_id = self.env['website.page'].search([('website_id', '=', self.id),
-                                                            ('key', '=', standard_homepage.key)])
+        homepage_page = Page.search([
+            ('website_id', '=', self.id),
+            ('key', '=', standard_homepage.key),
+        ])
+        if not homepage_page:
```

Comment:
homepage_view = self.with_context(website_id=self.id).viewref('website.homepage', raise_if_not_found=False)
if note homepage_page and homepage_view:

=> else crash possible

Reply:
Since we do just above
```python
standard_homepage = self.env.ref('website.homepage', raise_if_not_found=False)
if not standard_homepage:
    return False
```
We are sure that `website.homepage` exist.
So the condition would not change anything in this case as `viewref` fallback in generic.